Exiled.Events/Patches/Events/Player/Shoot.cs

```diff
@@ -29,276 +37,101 @@ namespace Exiled.Events.Patches.Events.Player
-                                        return false;
-                                    }
-
-                                    break;
-
-                                case HitBoxType.ARM:
-                                case HitBoxType.LEG:
-                                    num4 /= 2f;
-                                    break;
-                            }
-
-                            break;
-                    }
-
-                IL_6D1:
-                    num4 *= __instance.weapons[__instance.curWeapon].allEffects.damageMultiplier;
-                    num4 *= __instance.overallDamagerFactor;
-
-                    // >Exiled
-                    Log.Debug("Invoking late shoot.", Loader.ShouldDebugBeShown);
-
-                    var shotEventArgs = new ShotEventArgs(Player.Get(__instance.gameObject), target, hitboxType, num3, num4);
-
-                    Handlers.Player.OnShot(shotEventArgs);
-
-                    if (!shotEventArgs.CanHurt)
-                        return false;
-
-                    // <Exiled
-                    __instance._hub.playerStats.HurtPlayer(new PlayerStats.HitInfo(shotEventArgs.Damage, __instance._hub.LoggedNameFromRefHub(), DamageTypes.FromWeaponId(__instance.curWeapon), __instance._hub.queryProcessor.PlayerId), referenceHub.gameObject, false);
-                    __instance.RpcConfirmShot(true, __instance.curWeapon);
-                    __instance.PlaceDecal(true, new Ray(__instance.camera.position, dir), (int)referenceHub.characterClassManager.CurClass, num3);
-                    return false;
-                }
-                else
-                {
-                    if (target != null && hitboxType == HitBoxType.WINDOW && target.GetComponent<BreakableWindow>() != null)
-                    {
-                        float time = Vector3.Distance(__instance.camera.transform.position, target.transform.position);
-                        float damage = __instance.weapons[__instance.curWeapon].damageOverDistance.Evaluate(time);
-                        target.GetComponent<BreakableWindow>().ServerDamageWindow(damage);
-                        __instance.RpcConfirmShot(true, __instance.curWeapon);
-                        return false;
-                    }
-
-                    __instance.PlaceDecal(false, new Ray(__instance.camera.position, dir), __instance.curWeapon, 0f);
-                    __instance.RpcConfirmShot(false, __instance.curWeapon);
-                    return false;
-                }
-            }
-            catch (Exception e)
+                // Player.Get(this.gameObject)
+                new CodeInstruction(OpCodes.Ldarg_0).MoveLabelsFrom(newInstructions[index]),
+                new CodeInstruction(OpCodes.Callvirt, PropertyGetter(typeof(WeaponManager), nameof(WeaponManager.gameObject))),
```

Comment:
Does it work? The |gameObject` propery is from the `Component` and it should be `call` cuz it's `this` above.

Reply:
It uses callvirt in the base-game IL code, so that's why I used it here. I double checked and the event works as expected with Callvirt.

Reply:
You could even load the `this._hub` field instead of calling the `gameObject` getter.
I looked at `WeaponManager.CallCmdShoot` and the compiler uses `Call` to call `this.gameObject`.